src/Pingalot.Core/PingRequestAgent.cs

```diff
@@ -95,5 +120,31 @@ private static bool CheckBuffer(byte[] expected, byte[] actual)
 
 			return true;
 		}
+
+		private void SetupExportFile(string exportFile)
+		{
+			// open file and write out the csv file headers - just once
+			// we use append as file may already exist - thats ok still write to it
+			using (var stream = File.Open(exportFile, FileMode.Append))
+			using (var writer = new StreamWriter(stream))
+			using (var csv = new CsvWriter(writer, CultureInfo.InvariantCulture))
+			{
+				csv.WriteHeader<PingRequestExportModel>();
+				csv.NextRecord();
+			}
+		}
+
+		private void WriteRecordToExportFile(string exportFile, PingRequest pingRequest)
+		{
+			// write a single pingrequest record to export file
+			using (var stream = File.Open(exportFile, FileMode.Append))
+			using (var writer = new StreamWriter(stream))
+			using (var csv = new CsvWriter(writer, CultureInfo.InvariantCulture))
+			{
+				var singleExportablePingResult = new PingRequestExportModel(pingRequest);
+				csv.WriteRecord(singleExportablePingResult);
+				csv.NextRecord();
+			}
+		}
```

Comment:
While this approach works - opening and closing the file per request - it would be better to restructure the code to use a single stream/writer/csvwriter across the entire time. Not sure how best to do that off the top of my head right now but feel free to experiment with techniques to do so.

Reply:
@Turnerj I have had a play with this idea and I was able to get it to work, but, is it possible to have a single stream and be able to access the file externally with another program, on the file system? Across long ping sessions, I have found it useful to access the exported file in another app, like Notepad++ or BareTail, in order to see the detail of individual pings. This is possible when we open and close the file for each record. With a single stream I have tried flushing to disk and still the file is locked until the stream is closed.

Reply:
maybe another idea is to use a logger to push each ping request to disk....

Reply:
Had a quick look, we can set the `FileShare` for a `FileStream`: https://docs.microsoft.com/en-us/dotnet/api/system.io.fileshare?view=net-6.0

This will allow us to have an exclusive write while other programs can read. We can probably leverage that here.

Reply:
> Had a quick look, we can set the `FileShare` for a `FileStream`: https://docs.microsoft.com/en-us/dotnet/api/system.io.fileshare?view=net-6.0
> 
> This will allow us to have an exclusive write while other programs can read. We can probably leverage that here.

This is EXACTLY what i was looking for and my google fu was lacking..... :-( 
I have tested this and it works perfectly. Thanks
`using (var stream = File.Open(options.ExportFile, FileMode.Append,FileAccess.Write,FileShare.Read))`

WIP